src/containers/constructor-form-container/constructor-form-container.js

```diff
@@ -0,0 +1,256 @@
+import LoadingBar from '../../components/loading-bar';
+import CheckboxOptions from '../../components/checkbox-options';
+import LanguagePanel from '../../components/forms/language-panel';
+import ImageUploadPreviewContainer from '../image-upload-container/image-upload-previewContainer';
+import { config } from '../../configs';
+import MaterialsContainer from '../materials-container';
+import { useUnsavedChangesHandler } from '../../hooks/form-dialog/use-unsaved-changes-handler';
+import AdditionalPriceContainer from '../additional-price-container';
+
+import { materialSelector } from '../../redux/selectors/material.selectors';
+import { getMaterialsByPurpose } from '../../redux/material/material.actions';
+
+import useConstructorFormHandlers from '../../utils/use-constructor-form-handlers';
+import {
+  getDefaultPartItem,
+  getCheckboxOptions,
+  getPartItemInitialValues,
+  partItemColorsHandler,
+  getValidationSchema
+} from '../../utils/constructor-form-utils';
+import {
+  imagePreviewId,
+  defaultProps,
+  constructorObjectPropsTypes,
+  defaultPropTypes,
+  valuesPropTypes,
+  imagePropTypes
+} from './constructor-form.variables';
+
+const { IMG_URL } = config;
+const { SAVE_TITLE } = config.buttonTitles;
+const { languages, imagePrefix } = config;
+const { constructorItemLabels } = config.labels;
+
+const {
+  availableLabel,
+  uploadLabel,
+  additionalPriceContainer,
+  constructorItems
+} = constructorItemLabels;
+
+const { inputFields } = constructorItems.bottom;
+
+const ConstructorFormContainer = ({
+  part,
+  id,
+  edit,
+  partItemKey,
+  pathBack,
+  dispatchAction
+}) => {
+  const styles = useStyles();
+  const dispatch = useDispatch();
+  const partItem = part ?? getDefaultPartItem(partItemKey);
+  const { optionType } = partItem;
+  const { featuresLabels } = constructorItems[partItemKey];
+  const { materialsByPurpose, loading } = useSelector(materialSelector);
+  const materials = materialsByPurpose?.[partItemKey] || [];
+
+  const {
+    createPartItem,
+    setPartItemUpload,
+    partItemUpload,
+    setPartItemImage,
+    partItemImage,
+    colors,
+    setColors
+  } = useConstructorFormHandlers();
+
+  useEffect(() => {
+    dispatch(getMaterialsByPurpose([optionType]));
+  }, []);
+
+  useEffect(() => {
+    if (partItem?.images.thumbnail) {
+      setPartItemImage(`${imagePrefix}${partItem.images.thumbnail}`);
+    }
+  }, []);
+
+  const {
+    values,
+    handleSubmit,
+    handleChange,
+    handleBlur,
+    touched,
+    errors,
+    dirty,
+    isValid,
+    setFieldValue
+  } = useFormik({
+    validationSchema: getValidationSchema(optionType),
+    initialValues: getPartItemInitialValues(edit, IMG_URL, partItem),
+
+    onSubmit: () => {
+      const editAndUpload = edit && partItemUpload instanceof File;
+      const newPartItem = createPartItem(values);
+      const actionPayload = {};
+      actionPayload[partItemKey] = newPartItem;
+
+      if (edit) {
```

Comment:
Isn't this the same, but more concise?

      if (edit) {
        actionPayload.id = id;
      }

      if (editAndUpload) {
        actionPayload.image = partItemUpload;
      }

      dispatch(dispatchAction(actionPayload));